### DISCUSSION TITLE
Drop support for Python 3.8

### DISCUSSION THREAD

**Discussion by graingert** [encode/starlette#2820] [Ideas] [answered]
I recently ran into https://github.com/python/cpython/issues/69968 in https://github.com/encode/starlette/pull/2814

and would like to avoid a work-around.

edit: I just came up with a nicer work-around that's actually better than the code that needed the work-around: https://github.com/encode/starlette/pull/2814/files#diff-aca25e5f16c4fd49338ccdf3631f72455309335fee1e27f3d3b6016fa94ecedfR103-R110

**Comment by Kludex** [comment, marked as answer]
PR welcome.
